azurerm/data_source_compute_resource_sku.go

```diff
@@ -0,0 +1,442 @@
+
+	name := d.Get("name").(string)
+
+	resourceSkus, err := client.ListComplete(ctx)
+
+	if err != nil {
+		return fmt.Errorf("Error loading Compute Resource SKUs: %+v", err)
+	}
+
+	filteredResourceSkus := make([]compute.ResourceSku, 0)
+	for resourceSkus.NotDone() {
+		shouldInclude := false
+		sku := resourceSkus.Value()
+
+		if v, ok := d.GetOkExists("location"); ok {
+			if location := v.(string); location != "" {
+				if Contains(*sku.Locations, location) && *sku.Name == name {
+					shouldInclude = true
+				}
+			}
+		} else {
+			if *sku.Name == name {
+				shouldInclude = true
+			}
+		}
+
+		if shouldInclude {
+			filteredResourceSkus = append(filteredResourceSkus, sku)
+		}
+		err = resourceSkus.NextWithContext(ctx)
+		if err != nil {
+			return fmt.Errorf("Error loading Compute Resource SKUs: %+v", err)
+		}
+	}
+
+	d.SetId(time.Now().UTC().String())
+	d.Set("name", name)
+
+	log.Printf("[DEBUG] filteredResourceSkus: %+v\n", filteredResourceSkus)
+
+	if len(filteredResourceSkus) == 0 {
+		return fmt.Errorf("Error loading Resource SKU: could not find SKU '%s'. Invalid SKU Name or not valid in this subscription or location.", name)
+	} else if len(filteredResourceSkus) == 1 {
+		frs := filteredResourceSkus[0]
+
+		if resource_type := frs.ResourceType; resource_type != nil {
+			d.Set("resource_type", *frs.ResourceType)
+		}
+
+		if tier := frs.Tier; tier != nil {
+			d.Set("tier", *frs.Tier)
+		}
+
+		if size := frs.Size; size != nil {
+			d.Set("size", *frs.Size)
+		} else {
+			d.Set("size", "")
+		}
+
+		if family := frs.Family; family != nil {
+			d.Set("family", *frs.Family)
+		}
+
+		if kind := frs.Kind; kind != nil {
+			d.Set("kind", *frs.Kind)
+		} else {
+			d.Set("kind", "")
+		}
+
+		if capacity := frs.Capacity; capacity != nil {
+			d.Set("capacity", *frs.Capacity)
+		} else {
+			d.Set("capacity", "")
+		}
+
+		if frs.LocationInfo != nil {
+			locationInfo := flattenLocationInfo(frs.LocationInfo)
+			if err := d.Set("location_info", locationInfo); err != nil {
+				return fmt.Errorf("Error setting `location_info`: %+v", err)
+			}
+		}
+
+		if api_versions := frs.APIVersions; api_versions != nil {
+			d.Set("api_versions", *frs.APIVersions)
+		}
+
+		if frs.Costs != nil {
+			costs := flattenCosts(frs.Costs)
+			if err := d.Set("costs", costs); err != nil {
+				return fmt.Errorf("Error setting `costs`: %+v", err)
+			}
+		} else {
+			d.Set("costs", make([]int, 0))
+		}
+
+		if frs.Capabilities != nil {
+			capabilities := flattenCapabilities(frs.Capabilities)
+			if err := d.Set("capabilities", capabilities); err != nil {
+				return fmt.Errorf("Error setting `capabilities`: %+v", err)
+			}
+		}
+
+		if frs.Restrictions != nil {
+			restrictions := flattenRestrictions(frs.Restrictions)
+			if err := d.Set("restrictions", restrictions); err != nil {
+				return fmt.Errorf("Error setting `restrictions`: %+v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func Contains(a []string, x string) bool {
```

Comment:
Could we move this into the utils package? and rename it: `StringSliceContains()`